hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CombinedBlockCache.java

```diff
@@ -38,345 +32,15 @@
  * Metrics are the combined size and hits and misses of both caches.
  */
 @InterfaceAudience.Private
-public class CombinedBlockCache implements ResizableBlockCache, HeapSize {
-  protected final FirstLevelBlockCache l1Cache;
-  protected final BlockCache l2Cache;
-  protected final CombinedCacheStats combinedCacheStats;
+public class CombinedBlockCache extends CompositeBlockCache implements ResizableBlockCache {
```

Comment:
So compositeBlockCache is the base now. CombinedBlockCache is a type of it. good.